[cross language demo]go language deserialize demo

helintongh opened the pull request:
## Why

struct_pack support other language.

## What is changing

simple_serialize.cpp will serialize a struct named person and create a file which content is serialized string and it's name is example.txt.It also a demo which deserialize string to c language.

go_deserialize.go is a demo that reads this file(example.txt) , which deserializes the string assignment into the struct.

## Example

in build/example dirent, use bin cross_language_demo,generate a binary file example.txt. use go run o_deserialize.go.

helintongh committed:
[cross language demo]go language deserialize demo

File use:
simple_serialize.cpp will serialize struct person and create a file which content is serialized string and it's name is example.txt.It also a demo which deserialize string to c language.

go_deserialize.go is a demo that reads this file(example.txt)
, which deserializes the string assignment into the struct.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/yalantinglibs?pullRequest=128) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/yalantinglibs?pullRequest=128) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/yalantinglibs?pullRequest=128) it.</sub>

helintongh committed:
[cross language demo]go serialized string write to file and cpp read file deserialize to struct

go_deserialize will create a file named golang_serialize.txt.
read_from_other which source code is deserialize_from_other_language.cpp will read serialized string and deserialize string to cpp struct.

the struct person meta json is following:

{
   "hash_code": 2242444774,
   "struct_name": "person",
   "fileds": [
      {
         "field_name": "age",
         "field_type": "int32"
      },
      {
         "field_name": "name",
         "field_type": "string"
      }
   ]
}

PikachuHyA commented:
please sign CLA

PikachuHyA commented on src/struct_pack/examples/serialize/go_deserialize.go:
```diff
@@ -0,0 +1,80 @@
+package main
```
add ci for this demo is better

helintongh replied:
ok,I will change package name and add test for this demo
